### ISSUE TITLE
Very slow website load time when adguard is active

### ISSUE THREAD

**Issue by TommyKing** [AdguardTeam/AdguardForAndroid#4214]
As title said, when adguard is on very randomly, it will slow down connection to crawl, and when i turn it off, the connection is very fast
Adguard version : v4 nightly 35
Links example : https://www.techradar.com/news/i-tested-lgs-new-dolby-atmos-earbuds-against-airpods-and-theyre-wild
Filters
Easy List / Adguard base filter
Adguard tracking
Adguard social
Adguard Annoyance
AdblockID

[adguard_logs_4.0.77_1208_214650.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/9320722/adguard_logs_4.0.77_1208_214650.zip)

Here's adguard log with everything logged

Samething happened with stable.

**Comment by maxikuzmin** [comment]
@TommyKing it's not happening for me. We'll see what's wrong.

**Comment by TommyKing** [comment]
Thank you, but as i said it is random, but i am sure because of adguard. I have now turned off adguard for 2-3 days, and everything is fast

**Comment by ChibiChiii** [comment]
I seem to having the same problem. It becomes unbareable to keep adguard on. At first I thought it was a problem with my wifi connection because the problem only persisted when connected to the wifi. Turning the wifi on and off would fix it temporarily. However using several phones, one with adguard and one without I was able to determine that the wifi was working fine on a device that does not have adguard installed. The problem started occuring also on my windows 10 laptop with adguard installed. The connection will just stall and completely reset after like 1-2 minutes of extremely slow traffic. It's like a buffer is filling up and overloading or something. I can literally see the wifi disconnecting on my laptop, on my Android device I just get the error messages DNS cannot be found. Reconnecting to 4G or wifi completely fixes it. As I stated though, my husband does not use adguard neither on his windows surface nor his phone and he does not have the same issues. I disabled adguard protection and suddenly the problems with my internet connection were also gone. Turning adguard protection back on and within a few hours they were back. So it's safe to say it definitely is an issue with adguard, however not just the android app as the windows version has the same problem. It also only happens on wireless connections. I've used a lan connection on my desktop PC for years with adguard without a problem, I switched to a wireless network USB adapter, Asus Netgear, and the desktop pc has the same problem too now. The internet traffic becomes extremely slow when adguard is enabled.

**Comment by TommyKing** [comment]
@ChibiChiii Thank you for adding scenario, hopefully in my logs they will find the problem and fix it. I just can't use adguard for android if this keeps happening

**Comment by crsag** [comment]
I'm having same slowdown problem with adguard android.

**Comment by TommyKing** [comment]
> I'm having same slowdown problem with adguard android.

I think many of us got this problem. Hopefully they can fix this. I already uninstalled my adguard, becauae i dont even use it

**Comment by maxikuzmin** [comment]
@TommyKing @crsag @ChibiChiii  to troubleshoot this issue, we need to get additional logs.

Here's what we need you to do:

- Collect the debug log as it's [explained here](https://kb.adguard.com/en/android/solving-problems/log#debug);
- Remember the exact time when the issue was reproduced. We will need it to find the corresponding records in the log file;
- Additionally, capture the system bug report after reproducing the issue. [Here's the instruction](https://developer.android.com/studio/debug/bug-report);
- Send both files to devteam@adguard.com. Mention the Github issue number and the exact time when the issue was reproduced.

**Comment by ChibiChiii** [comment]
Dear Max,

I'm using Windows Adguard 7.9.1 and the collect the debug log on the link is hard to follow. I do not have an advanced settings in the left menu, only at the bottom of my settings and there is no logging level in these advanced settings.

Only Enable HAR writing, show Quic, and Block Java. The problem with this issue is that it randomly appears, sometimes it can take hours other times it's minutes. The HAR writing explains that it will slow down browsing significantly, which is the error we are trying to reproduce... so how are we supposed to remember the exact time the issue was produced if the logging will cause the exact same issue?

I'd love to help to find a solution to this problem as it does seem to happen to a lot of other people. Let me know how to do so please :)

**Comment by TommyKing** [comment]
> @TommyKing @crsag @ChibiChiii to troubleshoot this issue, we need to get additional logs.
> 
> Here's what we need you to do:
> 
>     * Collect the debug log as it's [explained here](https://kb.adguard.com/en/android/solving-problems/log#debug);
> 
>     * Remember the exact time when the issue was reproduced. We will need it to find the corresponding records in the log file;
> 
>     * Additionally, capture the system bug report after reproducing the issue. [Here's the instruction](https://developer.android.com/studio/debug/bug-report);
> 
>     * Send both files to [devteam@adguard.com](mailto:devteam@adguard.com). Mention the Github issue number and the exact time when the issue was reproduced.

I have sent you guys the logs even the time it happened to devteam@adguard.com and this thread link
No one replied to me. Support are so slow these days

**Comment by maxikuzmin** [comment]
@ChibiChiii everything is correct, as the issue is in AdGuard for Android repository - instructions to upload logs for Android. For Windows the instructions are as follows:

To troubleshoot this issue, we need to get the app logs.

Here's what we need you to do:

- Collect the debug log as it's [explained here](https://kb.adguard.com/en/windows/solving-problems/adguard-logs).
- Remember the exact time when the issue was reproduced. We will need it to find the corresponding records in the log file.
- Send the archive to devteam@adguard.com and mention this issue number in the subject.

**Comment by maxikuzmin** [comment]
@TommyKing indeed, saw your logs. Thanks, could you unload your settings, just in case? Send also to mail[devteam@adguard.com](mailto:devteam@adguard.com)

**Comment by TommyKing** [comment]
> @TommyKing indeed, saw your logs. Thanks, could you unload your settings, just in case? Send also to mail[devteam@adguard.com](mailto:devteam@adguard.com)

Just sent it

**Comment by maxikuzmin** [comment]
@ChibiChiii @TommyKing we see in the log that it does time out connections sometimes. There is nothing we can do about it. Are you sure that playback only happens with Adgard?

**Comment by ChibiChiii** [comment]
Positive, I've turned adguard off on my phone and haven't had the problem for 2 weeks now.

**Comment by TommyKing** [comment]
> Positive, I've turned adguard off on my phone and haven't had the problem for 2 weeks now.

Same i've turned off adguard for 3 weeks with no stalling issues

**Comment by maxikuzmin** [comment]
@TommyKing @ChibiChiii have you tried disabling DNS or changing it? Like change it to Google DNS?

**Comment by ChibiChiii** [comment]
My DNS is decided by my internet provider I don't have the ability to change it.

**Comment by TommyKing** [comment]
@maxikuzmin yes, i use my own dot. Which worked fine when i turned off adguard

**Comment by TommyKing** [comment]
@ChibiChiii could you try disabling phising and malware protection? Seems like my problem is alleviated when i disabled that feature, will need to test further

**Comment by maxikuzmin** [comment]
@ChibiChiii @TommyKing in DNS filtering -> Server selection -> try selecting Google DNS 
Or just try disabling DNS filtering

**Comment by TommyKing** [comment]
@maxikuzmin i disabled that all the time and use android private dns feature. I also tried to use dns.google on that , but the stalling still happened.
Note: i also disable browsing security from adguard before anyone ask

**Comment by TommyKing** [comment]
@maxikuzmin any update?

**Comment by maxikuzmin** [comment]
@TommyKing we're still figuring out

**Comment by ChibiChiii** [comment]
Hey guys, I have additional logs to add for you from my Windows Computer however the Zip file is 26 MB instead of 25 MB so it wouldn't let me mail or add them here in the github. Can you guys advice me how to get the file to you?

Best Regards,

Chi

**Comment by maxikuzmin** [comment]
@ChibiChiii google disc or one drive maybe?

**Comment by ChibiChiii** [comment]
Please download the zip here https://drive.google.com/file/d/1btmta0oJpHgyxnqc0ZvstCUtFLfSjgHN/view?usp=sharing

**Comment by maxikuzmin** [comment]
@ChibiChiii so is the issue on Android or Windows? The latest logs are from Windows, and the original issue is in the Android repository

**Comment by TommyKing** [comment]
@ChibiChiii i think you should open an issue in windows if your problem is on pc. This thread only for android, the devs are not the same, and it would be confusing

@maxikuzmin i think she said she had the same problem on windows, thats why she got the logs on windows. She should open a tracker on adguard on windows github.

**Comment by ChibiChiii** [comment]
> @ChibiChiii so is the issue on Android or Windows? The latest logs are from Windows, and the original issue is in the Android repository

The issue is on both devices, however I have removed Adguard from my Android Phone a few days after the issue was reported as it made my phone unbearably slow and constantly losing Wifi connection. Since Tommy already provided the logs for the Android device I figured I would supply the logs for the Windows device in addition since it appears to be the exact same problem.

**Comment by maxikuzmin** [comment]
@TommyKing @ChibiChiii we'll take a look at the logs, however it would be nice to get the android logs.

**Comment by TommyKing** [comment]
@maxikuzmin hopefully this issue can be fixed. Can't use adguard if this issue hasn't been fixed, it is so slow to browse

**Comment by maxikuzmin** [comment]
@ChibiChiii let me remind you that you sent logs from Windows system. Since the issue is in the Android repository, it would be nice to get the logs from Android.
@TommyKing you didn't manage to catch the problem again and record the logs?
Logs from the Windows system are still being examined

**Comment by ChibiChiii** [comment]
@maxikuzmin as I already explained I no longer use adguard on my phone so I cannot provide you with the Android logs. I have a license for 2 devices and am using it on both my desktop and my laptop. The Android Adgaurd was giving me too much trouble so I had to remove it.

**Comment by TommyKing** [comment]
> @ChibiChiii let me remind you that you sent logs from Windows system. Since the issue is in the Android repository, it would be nice to get the logs from Android. @TommyKing you didn't manage to catch the problem again and record the logs? Logs from the Windows system are still being examined

@maxikuzmin i think in the email i gave you a month ago i managed to get a video for you let me check my email. Yap checked my email, i have given you the video when it happened. 12 august 2022 even the log

**Comment by dvitaz** [comment]
i got mine working - what i had to do is forget my wifi and reconnect and that seemed to fix my issues

**Comment by ChibiChiii** [comment]
> i got mine working - what i had to do is forget my wifi and reconnect and that seemed to fix my issues

That's a temporary fix indeed. Works untill the next time it happens, just disconnecting and reconning works fine too no need to completely forget it.

**Comment by maxikuzmin** [comment]
@ChibiChiii @dvitaz still dealing with the situation. 
For how long does this temporary solution help?

**Comment by ChibiChiii** [comment]
Depending on the usage of the device, when I'm browsing the internet several hours at best. When I'm only using the device to chat for example on discord it lasts sometimes a day two at best. I sincerely feel like it's some sort of temp storage problem, like the filter from adguard fills up or something like that is that possible? I've tried to reproduce the problem on my phone lately but it's difficult to use the phone than and it's very slow especially when I'm using apps like Facebook or AlieXpress. I figure these apps normally have a lot of adds in them, so they must be on the heavier side to block all adds for adguard. Where as when I'm using Discord, that program merely uses the ocassional add. So there's much less traffic to block.

**Comment by TommyKing** [comment]
> @ChibiChiii @dvitaz still dealing with the situation. For how long does this temporary solution help?

after last update OS from Samsung and new build from ag android. i am seeing less lockup ( i haven't seen it in the past 5 days. it has improved alot), i will try monitoring this a bit more this past week. with wifi and mobile data

**Comment by maxikuzmin** [comment]
@TommyKing are the lockup still there?

**Comment by TommyKing** [comment]
nope. seems like some updates fixed it. adguard / samsung update. dont know which one

**Comment by maxikuzmin** [comment]
@ChibiChiii what about you?

**Comment by ChibiChiii** [comment]
> @ChibiChiii what about you?

I haven't had any hang ups on my windows device since a few weeks, currently not using the android version anymore because of a limited licence. But so far it seems to have solved.

**Comment by maxikuzmin** [comment]
@ChibiChiii @TommyKing I'm closing the issue. If the problem occurs again, please reopen the issue.
